lib/network/modules/components/nodes/util/CircleImageBase.js

```diff
@@ -10,10 +10,44 @@ class CircleImageBase extends NodeBase {
 
   setOptions(options, imageObj, imageObjAlt) {
     this.options = options;
-    this.setImages(imageObj, imageObjAlt);
+
+    //
+    // The issue here is that this setImages() is called twice on initialization:
+    //
+    // - First here, via the constructor call in parent NodeBase
+    // - Second in the constructor in derived Image or CircleImage.
+    //
+    // As an extra complication, this setOptions() is also called
+    // on an update to the node, where the two images are passed
+    // for Image and CircleImage (only).
+    //
+    // In the first case, imageObj and ImageObjAlt are always undefined;
+    // we use this to differentiate the calls.
+    // TODO: this is hacky, find a better solution
+    //
+    if (!(imageObj === undefined && imageObjAlt === undefined)) {
```

Comment:
Yes, plenty of risks. All you need to do is call `setOptions()` with a new image struct.

Thankfully, accepted PR #3053 addresses this issue, so all is well now. I'll just delete those comments now.